src/postamble_minimal.js

```diff
@@ -34,6 +34,15 @@ function initRuntime(asm) {
 #endif
 
   /*** RUN_GLOBAL_INITIALIZERS(); ***/
+
```

Comment:
Agreed, that is the type of scheme I was thinking about as well. I do not have time just now to start working on that, could this land like this in the meanwhile to get printf operational in minimal runtime?

Reply:
I worry about adding so much temporary complexity - that stuff accumulates.

With that said, I get where you are coming from. So if you think you can get to that in a matter of weeks, lgtm (please open an issue + assign yourself). If it's more a matter of being unsure when you can get to it, then let's discuss other options (I can try to find time to do it, and we can land this afterwards).

Reply:
I am sorry, I am not in a position to comfortably be able to make a promise for that kind of obligation. I hope you understand.

Would it be ok for this PR to land if the lines
```
  // TODO: the following line is a hack - remove it, and figure out a more generic "startup initialization stack" that is defined
  // by libraries at compile time.
#if LibraryManager.has('library_fs.js')
  FS.init();
#endif
#if LibraryManager.has('library_tty.js')
  TTY.init();
#endif
```
were removed to avoid adding temporary complexity?

Reply:
I don't understand, how can the PR land without that part - how would those initializers get called?

Reply:
Before this PR, printf is not working either, but attempting to run a FS-enabled page will crash at parsing since __ATINIT__ does not exist. With this PR if the above contentious block is omitted, at least parsing will not crash so pages can run at least if they skip printing at runtime.

Reply:
I see. Well, actually, after some more thinking, let's just land this as-is. I have a a good idea in my head about how to do that refactoring, I think, so I can sign up to do that myself, very soon. When I do that, I'll clean up this part of the code as part of that.

lgtm

Reply:
Thanks for the details! No, it's not the comment. My general worry is that accumulating small "odd" things adds to complexity, and then when we get a weird bug then figuring it out becomes much harder. So I'm trying to be careful about the overall trajectory of how we add complexity (odd stuff, special cases, more options, more code paths, etc.).

Specifically, refactoring this atinit code is something I've wanted to do for a while, and now seems like a good time as it will reduce the total complexity in that specific area to even better than before this PR.